### PR TITLE
Add reactions to Slack messages

### PR DESCRIPTION
**Pull request by mszostok**
## Description

Changes proposed in this pull request:

- [x] Add support both for Socket and Cloud Slack platforms
- [x] Add reactions both for typed messages add interactive events (button clicks)
  - [x] works in thread
- [x] we don't do that for messages  that are visible only to a user. Basically, Slack doesn't support that. However, this is OK for us as otherwise it will look wired e.g. for the `kubectl` command builder

## Testing

Install Socket Slack and run:
-  `@Botkube kubectl get pods -A`
- `@Botkube kubectl`
- `@Botkube help` and click some buttons.

## Related issue(s)

- https://github.com/kubeshop/botkube/issues/1126

### CODE REVIEW

**Comment by huseyinbabal** [comment]
@mszostok this works on socket slack, but couldn't make it work for cloud slack. I added `reactions:write` to `BotkubeDev` user already and reinstalled botkube playground channel. Do I need to reconnect to cloud workspace?

File: pkg/bot/slack_msg_tracker.go [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+import (
+	"fmt"
+
+	"github.com/sirupsen/logrus"
+	"github.com/slack-go/slack"
+)
+
+const (
+	msgReceivedEmoji  = "eyes"
+	msgProcessedEmoji = "heavy_check_mark"
```
Comment (huseyinbabal) [inline pkg/bot/slack_msg_tracker.go:12]: I would use following to show completed process
Screenshot 2023-07-20 at 10 59 35
If you are using dark mode, marked icon is almost invisible as follows
Screenshot 2023-07-20 at 11 00 36

File: pkg/bot/slack_socket.go [hunk 2]
```diff
@@ -216,6 +205,7 @@ func (b *SocketSlack) Start(ctx context.Context) error {
 						UserName:        userName,
 						CommandOrigin:   cmdOrigin,
 						State:           state,
+						EventTimeStamp:  callback.Message.Timestamp, // FIXME
```
Comment (huseyinbabal) [inline pkg/bot/slack_socket.go:208]: do we need to fix this in this PR?

Comment (mszostok) [reply pkg/bot/slack_socket.go:208]: the comment is  already outdated, I will remove it 👍

File: pkg/bot/slack_msg_tracker.go (hunk 1 above)
Comment (mszostok) [reply pkg/bot/slack_msg_tracker.go:12]: changed 👍 I also double-checked with Donna  and she also prefers "✅ "

**Review by huseyinbabal — APPROVED** [review]

**Pull request merged**